pkgs/development/tools/go-containerregistry/default.nix

```diff
@@ -0,0 +1,30 @@
+{ lib, buildGoModule, fetchFromGitHub, subPackages ? [ "cmd/crane" "cmd/gcrane" ] }:
+
+buildGoModule rec {
+  pname = "go-containerregistry";
+  version = "0.4.1";
+  src = fetchFromGitHub {
+    owner = "google";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-3mvGHAPKDUmrQkBKwlxnF6PG0ZpZDqlM9SMkCyC5ytE=";
+  };
+
+  vendorSha256 = null;
+
+  inherit subPackages;
+
+  buildFlagsArray = [
+    "-ldflags=-s -w -X github.com/google/go-containerregistry/cmd/crane/cmd.Version=${version} -X github.com/google/go-containerregistry/pkg/v1/remote/transport.Version=${version}"
```

Comment:
is there some clever way to get/use `goPackagePath` here?

Reply:
A let in could be used.